syft/generic/frameworks/hook/hook_args.py

```diff
@@ -101,6 +84,16 @@ def register_backward_func(new_backward_rules: Dict):
     backward_func = {**backward_func, **new_backward_rules}
 
 
+def register_ambiguous_method(*method):
+    global ambiguous_methods
```

Comment:
Why is it need to call global?

Reply:
Scoping in Python functions is one-way from global --> local, so modifications made to this variable will not be retained outside of this function unless we call `global` (or at least `nonlocal`).

Reply:
But to the larger question of why we need to do this whole registration business, it's so that we can register classes from outside of this module in these dictionaries/sets without importing such classes.

Reply:
> But to the larger question of why we need to do this whole registration business, it's so that we can register classes from outside of this module in these dictionaries/sets without importing such classes.

I totally understand and agree with this



> Scoping in Python functions is one-way from global --> local, so modifications made to this variable will not be retained outside of this function unless we call `global` (or at least `nonlocal`).

Have you tested it without global? I tried a simplified version of it, and it didn't seem necessary to turn this global

Reply:
Yes, it should work if you call them all in the same module, but `global` is necessary when calling these registration functions from a different module, which is the intended usage.